datacenter.go

```diff
@@ -79,6 +80,19 @@ func (c *Client) CreateDatacenter(dc Datacenter) (*Datacenter, error) {
 	return ret, err
 }
 
+// CreateDatacenterAndWait creates a data center, waits for the request to finish and returns a refreshed
+// result.
+func (c *Client) CreateDatacenterAndWait(ctx context.Context, dc Datacenter) (*Datacenter, error) {
+	rsp, err := c.CreateDatacenter(dc)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.WaitTillProvisionedOrCanceled(ctx, rsp.Headers.Get("location")); err != nil {
+		return nil, err
```

Comment:
You should return the created objects when wait fails to at least preserve the resource id

Reply:
good point